pkgs/development/python-modules/dxpy/default.nix

```diff
@@ -0,0 +1,50 @@
+{
+  lib,
+  buildPythonPackage,
+  pythonOlder,
+  fetchgit,
+
+  # runtime dependencies
+  certifi,
+  urllib3,
+  python-dateutil,
+  websocket-client,
+  argcomplete,
+  psutil,
+}:
+
+buildPythonPackage rec {
+  pname = "dxpy";
+  version = "0.377.0";
+
+  disabled = pythonOlder "3.8";
+
+  src = "${fetchgit {
+    url = "https://github.com/dnanexus/dx-toolkit.git";
+    rev = "v0.377.0";
+    hash = "sha256-xcUSl5C9XKFc0TQh7wJPeTP7syZCwU5BWo55mtSBdN0=";
+  }}/src/python";
+
+  dependencies = [
+    certifi
+    urllib3
+    python-dateutil
+    websocket-client
+    argcomplete
+    psutil
+  ];
+
+  doCheck = false;
```

Comment:
There are tests https://github.com/dnanexus/dx-toolkit/tree/master/src/python/test

To test whether it should be turned off, please add a comment above the `doCheck`.